api/src/org/apache/cloudstack/api/command/admin/account/CreateAccountCmd.java

```diff
@@ -196,5 +204,8 @@ private void validateParams() {
         if(StringUtils.isEmpty(getPassword())) {
             throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Empty passwords are not allowed");
         }
+        if (getAccountType() == null && getRoleId() == null) {
```

Comment:
Fixed the roleId < 1L issue.
No, here the check is for maintaining backward compatibility. The accounttype used to be a 'required' param, but now it's optional, and so is 'roleid'; this check is to ensure that at least one of the two are provided